tobac/utils/internal.py

```diff
@@ -454,16 +455,21 @@ def find_vertical_axis_from_coord(variable_cube, vertical_coord="auto"):
     ValueError
         Raised if the vertical coordinate isn't found in the cube.
     """
-
-    list_coord_names = [coord.name() for coord in variable_cube.coords()]
+    list_vertical = [
+        "z",
+        "model_level_number",
+        "altitude",
+        "geopotential_height",
+    ]
+
+    if isinstance(variable_cube, iris.cube.Cube):
+        list_coord_names = [coord.name() for coord in variable_cube.coords()]
+    elif isinstance(variable_cube, xr.Dataset) or isinstance(
+        variable_cube, xr.DataArray
+    ):
+        list_coord_names = variable_cube.coords
```

Comment:
**Suggestion:** Should we wrap this function with `@irispandas_to_xarray` to avoid checking internally?

Reply:
Doing this caused issues with some of the other functions, so I think debugging/updating this is outside the scope of this PR.